### PR TITLE
feat: Migration guide

### PR DESCRIPTION
**Pull request by emostov**
Not sure if we want this in the repo - but posting here as a starting point.

### CODE REVIEW

File: MIGRATION_GUIDE.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
+# v0.x.x &rightarrow; v1.0.0-beta.0
+
+Full documentation for v1 endpoints is available here: https://paritytech.github.io/substrate-api-sidecar/dist/
+
+For most users, the major bump will require transitioning all the endpoint paths
+they use to an updated version. The v1.0.0.beta.0 release contains both the old
```
Comment (joepetrowski) [inline MIGRATION_GUIDE.md:6]:
```suggestion
that they use to an updated version. The v1.0.0.beta.0 release contains both the old
```

File: MIGRATION_GUIDE.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+and new endpoints to help facilitate the transition.
+
+**`substrate-api-sidecar` releases after Friday, October 2nd, 2020, will not
```
Comment (joepetrowski) [inline MIGRATION_GUIDE.md:9]: This might be a little aggressive on time. Maybe do one more week.

File: MIGRATION_GUIDE.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+make any gurantees to include deprecated endpoints.** It is important that users
+finishing transitioning to the new endpoints by this date so they are ready for
+any subsequent emergency updates.
+
+Below is a table that is intended to give an overview of what the mapping of
+deprecated endpoints to there v1 equivalents. Many of the new
```
Comment (joepetrowski) [inline MIGRATION_GUIDE.md:15]:
```suggestion
Below is a table that is intended to give an overview of the mapping from
deprecated endpoints to their v1 equivalents. Many of the new
```

File: MIGRATION_GUIDE.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+endpoints have unlisted options for query params.
```
Comment (joepetrowski) [inline MIGRATION_GUIDE.md:16]: Add a note that some of the response objects are different too. Or maybe add a symbol on the v1 side of the table, like `*` to indicate that the response has a breaking change.

Edit: Oh, you cover it after

File: MIGRATION_GUIDE.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+| `/staking-info/{blockId}`         | `/pallets/staking/progress?at={blockId}`          |
+| (None)                            | `/pallets/{palletId}/storage/{storageItemId}`     |
+| (None)                            | `/node/network`                                   |
+| (None)                            | `/node/version`                                   |
+| (None)                            | `/node/transaction-pool`                          |
+| `/`                               | `/`
+
+**N.B.** the `/blocks` endpoint sees two significant breaking changes:
+
+1) extrinsic and event methods names no longer are a string of the form
```
Comment (joepetrowski) [inline MIGRATION_GUIDE.md:51]:
```suggestion
1) extrinsic and event method names are no longer a string of the form
```

File: MIGRATION_GUIDE.md [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+`pallet.method`, instead they are an object of the form
+`{ pallet: string, method: string }`.
+
+2) Unsigned extrinsics and inherents will have `null` for nonce and tip, while
+`paysFee` will always be `false`. Previously the latter two fields where 0, while
```
Comment (joepetrowski) [inline MIGRATION_GUIDE.md:56]:
```suggestion
`paysFee` will always be `false`. Previously the latter two fields were 0, while
```

File: MIGRATION_GUIDE.md [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+`paysFee` was often `true`. (#274)
+
+There are some other changes throughout the API, including some slightly updated
+error messages which have not yet been updated in the new docs UI.
```
Comment (joepetrowski) [inline MIGRATION_GUIDE.md:60]:
```suggestion
error messages that have not yet been updated in the new docs UI.
```

File: MIGRATION_GUIDE.md [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+
+Full documentation for v1 endpoints is available here: https://paritytech.github.io/substrate-api-sidecar/dist/
+
+For most users, the major bump will require transitioning all the endpoint paths
+they use to an updated version. The v1.0.0.beta.0 release contains both the old
+and new endpoints to help facilitate the transition.
+
+**`substrate-api-sidecar` releases after Friday, October 2nd, 2020, will not
+make any gurantees to include deprecated endpoints.** It is important that users
+finishing transitioning to the new endpoints by this date so they are ready for
```
Comment (dvdplm) [inline MIGRATION_GUIDE.md:11]:
```suggestion
complete the transition to the new endpoints by this date so they are ready for
```

File: MIGRATION_GUIDE.md (hunk 3 above)
Comment (dvdplm) [inline MIGRATION_GUIDE.md:15]:
```suggestion
Below is a table that gives an overview of what the mapping from
deprecated endpoints to their v1 equivalents. Many of the new
```

File: MIGRATION_GUIDE.md (hunk 4 above)
Comment (dvdplm) [reply MIGRATION_GUIDE.md:16]: "unlisted"? Can you elaborate on what that means?

**Review by dvdplm — CHANGES_REQUESTED** [review]

File: MIGRATION_GUIDE.md (hunk 2 above)
Comment (emostov) [reply MIGRATION_GUIDE.md]: See updated. I moved it back to October 16th.

File: MIGRATION_GUIDE.md (hunk 4 above)
Comment (emostov) [reply MIGRATION_GUIDE.md]: I removed "unlisted" and rephrased it to basically say "go look at the docs for more info on query params".

File: MIGRATION_GUIDE.md [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+| (None)                            | `/pallets/{palletId}/storage/{storageItemId}`     |
+| (None)                            | `/node/network`                                   |
+| (None)                            | `/node/version`                                   |
+| (None)                            | `/node/transaction-pool`                          |
+| `/`                               | `/`
+
+**N.B.** the `/blocks` endpoint sees two significant breaking changes:
+
+1) extrinsic and event method names are no longer a string of the form
+`pallet.method`, instead they are an object of the form
```
Comment (joepetrowski) [inline MIGRATION_GUIDE.md:52]:
```suggestion
1) Extrinsic and event method names are no longer a string of the form
`pallet.method`; instead they are an object of the form
```

**Review by joepetrowski — APPROVED** [review]

**Pull request merged**